srtcore/channel.cpp

```diff
@@ -90,86 +88,92 @@ m_iIpV6Only(-1)
 {
 }
 
-CChannel::CChannel(int version):
-m_iIPversion(version),
-m_iSocket(),
-#ifdef SRT_ENABLE_IPOPTS
-m_iIpTTL(-1),
-m_iIpToS(-1),
-#endif
-m_iSndBufSize(65536),
-m_iRcvBufSize(65536),
-m_iIpV6Only(-1),
-m_BindAddr(version)
+CChannel::~CChannel()
 {
-   SRT_ASSERT(version == AF_INET || version == AF_INET6);
-   m_iSockAddrSize = (AF_INET == m_iIPversion) ? sizeof(sockaddr_in) : sizeof(sockaddr_in6);
 }
 
-CChannel::~CChannel()
+void CChannel::createSocket(int family)
 {
+    // construct an socket
+    m_iSocket = ::socket(family, SOCK_DGRAM, IPPROTO_UDP);
+
+#ifdef _WIN32
+    int invalid = INVALID_SOCKET;
+#else
+    int invalid = -1;
+#endif
+
+    if (m_iSocket == invalid)
+        throw CUDTException(MJ_SETUP, MN_NONE, NET_ERROR);
+
+    if ((m_iIpV6Only != -1) && (family == AF_INET6)) // (not an error if it fails)
+        ::setsockopt(m_iSocket, IPPROTO_IPV6, IPV6_V6ONLY, (const char*)(&m_iIpV6Only), sizeof(m_iIpV6Only));
+
 }
 
-void CChannel::open(const sockaddr* addr)
+void CChannel::open(const sockaddr_any& addr)
 {
-   // construct an socket
-   m_iSocket = ::socket(m_iIPversion, SOCK_DGRAM, 0);
+    createSocket(addr.family());
+    socklen_t namelen = addr.size();
 
-   #ifdef _WIN32
-      if (INVALID_SOCKET == m_iSocket)
-   #else
-      if (m_iSocket < 0)
-   #endif
-      throw CUDTException(MJ_SETUP, MN_NONE, NET_ERROR);
+    if (::bind(m_iSocket, &addr.sa, namelen) == -1)
+        throw CUDTException(MJ_SETUP, MN_NORES, NET_ERROR);
 
-   if ((m_iIpV6Only != -1) && (m_iIPversion == AF_INET6)) // (not an error if it fails)
-      ::setsockopt(m_iSocket, IPPROTO_IPV6, IPV6_V6ONLY, (const char*)(&m_iIpV6Only), sizeof(m_iIpV6Only));
+    m_BindAddr = addr;
+    LOGC(mglog.Debug, log << "CHANNEL: Bound to local address: " << SockaddrToString(m_BindAddr));
```

Comment:
Alignment

Reply:
I can't see any wrong formatting here. To be sure, I reformatted this in vim set up for tab-to-spaces and 4-spaced alignment and nothing was changed.